http-client/src/main/java/io/micronaut/http/client/netty/FullNettyClientHttpResponse.java

```diff
@@ -221,19 +220,14 @@ private <T> Optional<T> convertByteBuf(ByteBuf content, Argument<T> type) {
                 MessageBodyReader<T> r = reader.get();
                 MediaType ct = contentType.get();
                 if (r.isReadable(type, ct)) {
-                    return Optional.of(r.read(type, ct, headers, NettyByteBufferFactory.DEFAULT.wrap(content.retainedSlice())));
+                    return Optional.of(r.read(type, ct, headers, NettyByteBufferFactory.DEFAULT.wrap(unpooledContent)));
                 }
             }
         } else if (LOG.isTraceEnabled()) {
             LOG.trace("Missing or unknown Content-Type received from server.");
         }
         // last chance, try type conversion
-        return conversionService.convert(content, ByteBuf.class, type);
-    }
-
-    @Override
-    public void onComplete() {
-        this.complete = true;
```

Comment:
oh, complete was unused? :smile:

Reply:
Yeah, the field was highlighted as not used